src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingCustomShardAllocationSpec.cs

```diff
@@ -0,0 +1,362 @@
+
+        [MultiNodeFact]
+        public void Cluster_sharding_with_custom_allocation_strategy_specs()
+        {
+            Cluster_sharding_with_custom_allocation_strategy_should_setup_shared_journal();
+            Cluster_sharding_with_custom_allocation_strategy_should_use_specified_region();
+            Cluster_sharding_with_custom_allocation_strategy_should_rebalance_specified_shards();
+        }
+
+        public void Cluster_sharding_with_custom_allocation_strategy_should_setup_shared_journal()
+        {
+            // start the Persistence extension
+            Persistence.Persistence.Instance.Apply(Sys);
+            RunOn(() =>
+            {
+                Persistence.Persistence.Instance.Apply(Sys).JournalFor("akka.persistence.journal.MemoryJournal");
+            }, _config.First);
+            EnterBarrier("persistence-started");
+
+            RunOn(() =>
+            {
+                Sys.ActorSelection(Node(_config.First) / "system" / "akka.persistence.journal.MemoryJournal").Tell(new Identify(null));
+                var sharedStore = ExpectMsg<ActorIdentity>(TimeSpan.FromSeconds(10)).Subject;
+                sharedStore.Should().NotBeNull();
+
+                MemoryJournalShared.SetStore(sharedStore, Sys);
+            }, _config.First, _config.Second);
+            EnterBarrier("after-1");
+
+            RunOn(() =>
+            {
+                //check persistence running
+                var probe = CreateTestProbe();
+                var journal = Persistence.Persistence.Instance.Get(Sys).JournalFor(null);
+                journal.Tell(new Persistence.ReplayMessages(0, 0, long.MaxValue, Guid.NewGuid().ToString(), probe.Ref));
+                probe.ExpectMsg<Persistence.RecoverySuccess>(TimeSpan.FromSeconds(10));
+            }, _config.First, _config.Second);
+            EnterBarrier("after-1-test");
+        }
+
+        public void Cluster_sharding_with_custom_allocation_strategy_should_use_specified_region()
+        {
+            Within(TimeSpan.FromSeconds(10), () =>
+            {
+                Join(_config.First, _config.First);
+
+                RunOn(() =>
+                {
+                    _allocator.Value.Tell(new UseRegion(_region.Value));
+                    ExpectMsg<UseRegionAck>();
+                    _region.Value.Tell(1);
+                    ExpectMsg(1);
+                    LastSender.Path.Should().Be(_region.Value.Path / "1" / "1");
+                }, _config.First);
+                EnterBarrier("first-started");
+
+                Join(_config.Second, _config.First);
+
+                _region.Value.Tell(2);
+                ExpectMsg(2);
+
+                RunOn(() =>
+                {
+                    LastSender.Path.Should().Be(_region.Value.Path / "2" / "2");
+                }, _config.First);
+                RunOn(() =>
+                {
+                    LastSender.Path.Should().Be(Node(_config.First) / "user" / "sharding" / "Entity" / "2" / "2");
```

Comment:
because `ClusterShardingGuardian` was created as user actor and not as system actor. Shall I change this?